examples/passing/2288.purs

```diff
@@ -0,0 +1,19 @@
+module Main where
+
+import Prelude
+import Control.Monad.Eff
+import Control.Monad.Eff.Console
+import Data.Array
+import Data.Array.Partial as P
+import Partial.Unsafe
+
+length :: forall a. Array a -> Int
+length = go 0 where
+  go acc arr =
+    if null arr
+    then acc
+    else go (acc + 1) (unsafePartial P.tail arr)
+
+main = do
+  logShow (length [1, 2, 3])
```

Comment:
Does it make sense to have either this test or another test loop with input large enough that it would have blow the stack if we didn't optimize?